File: backend/app/schemas/preferences.py
from enum import StrEnum

from pydantic import BaseModel


class ThemeMode(StrEnum):
    light = "light"
    dark = "dark"


class ThemeAccent(StrEnum):
    teal = "teal"
    blue = "blue"
    rose = "rose"


class ThemeDensity(StrEnum):
    comfortable = "comfortable"
    compact = "compact"


class ThemePreferences(BaseModel):
    mode: ThemeMode = ThemeMode.light
    accent: ThemeAccent = ThemeAccent.teal
    density: ThemeDensity = ThemeDensity.comfortable

File: frontend/src/api/preferences.ts
import { api } from "./client";

export interface ThemePreferences {
  mode: "light" | "dark";
  accent: "teal" | "blue" | "rose";
  density: "comfortable" | "compact";
}

export async function getPreferences(): Promise<ThemePreferences> {
  return api<ThemePreferences>("/api/v1/account/preferences");
}

export async function updatePreferences(
  prefs: ThemePreferences,
): Promise<ThemePreferences> {
  return api<ThemePreferences>("/api/v1/account/preferences", {
    method: "PUT",
    body: JSON.stringify(prefs),
  });
}
